src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/RequestsForPlacementTest.kt

```diff
@@ -0,0 +1,214 @@
+            .expectStatus()
+            .isNotFound
+        }
+      }
+    }
+
+    @Test
+    fun `Get a Request for Placement for an application that is not the application it belongs to returns a 404 response`() {
+      `Given a User` { user, jwt ->
+        `Given an Application`(user) { wrongApplication ->
+          `Given a Placement Request`(
+            placementRequestAllocatedTo = null,
+            assessmentAllocatedTo = user,
+            createdByUser = user,
+          ) { placementRequest, _ ->
+            webTestClient.get()
+              .uri("/applications/${wrongApplication.id}/requests-for-placement/${placementRequest.id}")
+              .header("Authorization", "Bearer $jwt")
+              .exchange()
+              .expectStatus()
+              .isNotFound
+          }
+        }
+      }
+    }
+
+    @Test
+    fun `Get a Request for Placement for an application returns a 200 response with the expected placement request`() {
+      `Given a User` { user, jwt ->
+        `Given an Assessment for Approved Premises`(
+          allocatedToUser = null,
+          createdByUser = user,
+        ) { assessment, application ->
+          val postCodeDistrict = postCodeDistrictFactory.produceAndPersist()
+
+          val placementRequirements = placementRequirementsFactory.produceAndPersist {
+            withApplication(application)
+            withAssessment(assessment)
+            withPostcodeDistrict(postCodeDistrict)
+            withEssentialCriteria(listOf())
+            withDesirableCriteria(listOf())
+          }
+
+          val placementRequest = placementRequestFactory.produceAndPersist {
+            withApplication(application)
+            withAssessment(assessment)
+            withPlacementRequirements(placementRequirements)
+            withCreatedAt(OffsetDateTime.now().roundNanosToMillisToAccountForLossOfPrecisionInPostgres())
+          }
+
+          webTestClient.get()
+            .uri("/applications/${application.id}/requests-for-placement/${placementRequest.id}")
+            .header("Authorization", "Bearer $jwt")
+            .exchange()
+            .expectStatus()
+            .isOk
+            .expectBody()
+            .json(
+              objectMapper.writeValueAsString(
+                requestForPlacementTransformer.transformPlacementRequestEntityToApi(placementRequest),
+              ),
+            )
+        }
+      }
+    }
+
+    @Test
+    fun `Get a Request for Placement for an application returns a 200 response with the expected placement application`() {
+      `Given a User` { user, jwt ->
+        val schema = approvedPremisesPlacementApplicationJsonSchemaEntityFactory.produceAndPersist()
+
+        `Given a Placement Application`(createdByUser = user, schema = schema) { placementApplication ->
+          webTestClient.get()
+            .uri("/applications/${placementApplication.application.id}/requests-for-placement/${placementApplication.id}")
```

Comment:
are we still planning to implement a composite ID e.g.

- `.../requests-for-placement/placement_application:bc12b7e8-a404-4216-97ad-1b18199e0602`
- `.../requests-for-placement/placement_request:49f282ad-ff82-4747-8a7d-9c30407c594e`

I believe we thought that this would be helpful:

- to the controller or service handling the request (perhaps `RequestForPlacementService. getRequestForPlacement()`) and also, perhaps especially
- to humans when debugging / developing

See https://dsdmoj.atlassian.net/browse/APS-587?focusedCommentId=444903